src/codegen/compile.ml

```diff
@@ -3912,22 +3705,19 @@ module BigNumLibtommath : BigNumType = struct
         then []
         else
           let (a, b) = Big_int.quomod_big_int n twoto28 in
-          [ Big_int.int32_of_big_int b ] @ go a
+          [ compile_unboxed_const (Big_int.int32_of_big_int b) ] @ go a
       in go n
     in
     (* how many 32 bit digits *)
     let size = Int32.of_int (List.length limbs) in
 
     (* cf. mp_int in tommath.h *)
-    let ptr = Tagged.shared_static_obj env Tagged.BigInt StaticBytes.[
-      I32 size; (* used *)
-      I32 size; (* size; relying on Heap.word_size == size_of(mp_digit) *)
-      I32 sign;
-      I32 0l; (* dp; this will be patched in BigInt::mp_int_ptr in the RTS when used *)
-      i32s limbs
-
-    ] in
-    ptr
+    E.SharedObject (fun env -> Tagged.obj env Tagged.BigInt ([
+      compile_unboxed_const size; (* used *)
+      compile_unboxed_const size; (* size; relying on Heap.word_size == size_of(mp_digit) *)
+      compile_unboxed_const sign;
+      compile_unboxed_const 0l; (* dp; this will be patched in BigInt::mp_int_ptr in the RTS when used *)
+    ] @ limbs))
```

Comment:
Whatif there are a lot of limbs? The wasm stack could get deep. Is that a problem?